larpandora/LArPandoraInterface/LArPandoraInput.cxx

```diff
@@ -226,6 +230,34 @@ void LArPandoraInput::CreatePandoraDetectorGaps(const Settings &settings, const
     {
         PandoraApi::Geometry::LineGap::Parameters parameters;
 
+        if (isDualPhase)
+        {
+            const bool  isGapInU ((std::fabs(gap.GetY2() - gap.GetY1()) > gap.GetMaxGapSize())); //Could have chosen Z here, resulting in switching Y<->Z and U<->V in the try{...} block below
+
+            try
+            {
+                parameters.m_lineGapType = ( isGapInU ? pandora::TPC_WIRE_GAP_VIEW_U : pandora::TPC_WIRE_GAP_VIEW_V); //If gapSizeY is too large then the gap is in Z, therefore should be in kU (i.e. kZ)
+                parameters.m_lineStartX = gap.GetX2();
+                parameters.m_lineEndX = gap.GetX1();
+                parameters.m_lineStartZ = ( isGapInU ? gap.GetZ1() : gap.GetY1());
+                parameters.m_lineEndZ = ( isGapInU ? gap.GetZ2() : gap.GetY2());
+            }
+            catch (const pandora::StatusCodeException &)
+            {
+                mf::LogWarning("LArPandora") << "CreatePandoraDetectorGaps - invalid line gap parameter provided, all assigned values must be finite, line gap omitted " << std::endl;
+                continue;
+            }
+            try
+            {
+                PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, PandoraApi::Geometry::LineGap::Create(*pPandora, parameters));
+            }
+            catch (const pandora::StatusCodeException &)
+            {
+                mf::LogWarning("LArPandora") << "CreatePandoraDetectorGaps - unable to create line gap, insufficient or invalid information supplied " << std::endl;
+                continue;
+            }
+            continue; //No drift gaps in DP
+        }
```

Comment:
Using exceptions for regular flow control is quite expensive.  A better approach is to use the direct API:
```.cpp
auto const rc = PandoraApi::Geometry::LineGap::Create(*pPandora, parameters);
if (rc != pandora::STATUS_CODE_SUCCESS) {
  mf::LogWarning("LArPandora") << ...;
  continue;
}
```